### ISSUE TITLE
[java] GuardLogStatement: false negative with `this` pointer

### ISSUE THREAD

**Issue by mohui1999** [pmd/pmd#4376]
**Affects PMD Version:** 6.52.0
 
**Rule:** [GuardLogStatement](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#guardlogstatement)

**Description:**
When using ` LOGGER.severe("……");`, PMD report with rule **GuardLogStatement**. While using `this.LOGGER.severe("……")`, it does not report. It may be a  false negative.

**Code Sample demonstrating the issue:**

Maybe a false negative:
```java
import java.util.logging.Logger;
import java.util.logging.Level;

public class Foo {

    Logger LOGGER;

    public void foo() {
        this.LOGGER.severe("This is a severe message" + this + " and concat");   // NOT report here
    }
}

```
Correctly reported code:

```java
import java.util.logging.Logger;
import java.util.logging.Level;

public class Foo {
    Logger LOGGER;

    public void foo() {
        LOGGER.severe("This is a severe message" + this + " and concat");   // report here
    }
}
```

**Expected outcome:**
```
GuardLogStatement:        Logger calls should be surrounded by log level guards.
```

PMD should report a violation at line 9, but doesn't. This is a false negative.

**Running PMD through:** *[CLI]*

**Comment by mohui1999** [comment]
similar to PMD[#2543 ](https://github.com/pmd/pmd/issues/2543)
with `this`:
<img src="https://user-images.githubusercontent.com/32874764/215805912-5bd47f65-b3f1-440f-87fc-e1655b05f8ff.png" width=30%>
without `this`:
<img src="https://user-images.githubusercontent.com/32874764/215806152-0bf366a4-9417-417c-8eb1-ba19cfc90aff.png" width=30%>

**Fix suggestion:**
In GuardLogStatement.java 

If meeting the case shown in this issue (e.g., `this.LOGGER.foo`), it can use `primary.getChild(2)` to find the wanted primarySuffix.

```
    ASTPrimarySuffix primarySuffix = (ASTPrimarySuffix) primary.getChild(1);
            //if meeting the case shown in this issue (e.g., this.LOGGER.foo)
            //primarySuffix = (ASTPrimarySuffix) primary.getChild(2)
                if (hasStringConcatenationWithVars(primarySuffix)
                        || hasArgumentWithMethodCall(primarySuffix)) {
                    addViolation(data, node);
                }
            }
```

**Comment by mohui1999** [comment]
Fixed in PMD 7.0
